telluric/georaster.py

```diff
@@ -740,10 +740,10 @@ def empty_from_roi(cls, roi=None, resolution=None,
                                        dtype=dtype, shape=shape, ul_corner=ul_corner, raster_cls=cls)
 
     def _cleanup(self):
-        for f in self._opened_files:
-            f.close()
-
-        self._opened_files = []
+        if self._opened_files:
+            for f in list(self._opened_files):
+                f.close()
+                self._opened_files.pop(0)
```

Comment:
a way to avoid to ionvalidate iterator could be to pop the last one in a reversed order list... btw also this solution work